src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -94,25 +112,133 @@ static void assumeDocker() throws Exception {
+                s = prepareAgentDocker();
+                j.jenkins.addNode(s);
+                j.waitOnline(s);
+                break;
+            default:
+                fail("Unknown enum value: " + platform);
+                break;
+        }
+        ws = s.getWorkspaceRoot().child("ws");
+        launcher = s.createLauncher(listener);
+    }
 
-    @Before public void vars() {
-        listener = StreamTaskListener.fromStdout();
-        ws = j.jenkins.getRootPath().child("ws");
-        launcher = j.jenkins.createLauncher(listener);
+    private Slave prepareAgentDocker() throws Exception {
+        switch(platform) {
+            case SLIM:
+            case ALPINE:
+            case CENTOS:
+            case UBUNTU:
+                return prepareDockerPlatforms();
+            case SIMPLE:
+                return prepareAgentCommandLauncher();
+            default:
+                fail("Unknown test platform: " + platform);
+                return null;
+        }
     }
 
-    @Test
-    public void smokeTest() throws Exception {
-        Controller c = new BourneShellScript("echo hello world").launch(new EnvVars(), ws, launcher, listener);
+    private Slave prepareDockerPlatforms() throws Exception {
+        DockerContainer container = null;
+        String customJavaPath = null;
+        switch (platform) {
+            case SLIM:
+                container = dockerSlim.get();
+                customJavaPath = SlimFixture.SLIM_JAVA_LOCATION;
+                break;
+            case ALPINE:
+                container = dockerAlpine.get();
+                break;
+            case CENTOS:
+                container = dockerCentOS.get();
+                break;
+            case UBUNTU:
+                container = dockerUbuntu.get();
+                break;
+            default:
+                fail("Unknown enum value: " + platform);
+                break;
+        }
+        SystemCredentialsProvider.getInstance().setDomainCredentialsMap(Collections.singletonMap(Domain.global(), Collections.<Credentials>singletonList(new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "test", null, "test", "test"))));
+        SSHLauncher sshLauncher = new SSHLauncher(container.ipBound(22), container.port(22), "test");
+        sshLauncher.setJavaPath(customJavaPath);
+        return new DumbSlave("docker", "/home/test", sshLauncher);
+    }
+
+    private Slave prepareAgentCommandLauncher() throws Exception{
+        // counter used to prevent name smashing when a docker container from the previous test is
+        // still shutting down but the new test container is being spun up. Seems more ideal than adding a wait.
+        String name = "docker";
+        String agent = "agent-" + counter++;
+        String remoteFs = "/home/jenkins/" + agent;
+        String dockerRunSimple = String.format("docker run -i --rm --name %s jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar", agent);
```

Comment:
check https://hub.docker.com/r/jenkins/slave/tags for updates

```suggestion
        String dockerRunSimple = String.format("docker run -i --rm --name %s jenkins/slave:3.29-2 java -jar /usr/share/jenkins/slave.jar", agent);
```